cmd/subctl/upgrade.go

```diff
@@ -0,0 +1,271 @@
+	ctx := context.TODO()
+
+	status.Start("Starting upgrade process")
+	defer status.End()
+
+	brokerObj, brokerFound, err := getBroker(clusterInfo.RestConfig, constants.DefaultBrokerNamespace)
+	if err != nil {
+		return err
+	}
+
+	if brokerFound {
+		// Role updates are part of Broker redeploy
+		err = upgradeBroker(ctx, clusterInfo, status, brokerObj.Namespace, brokerObj.Spec)
+		if err != nil {
+			return status.Error(err, "Error upgrading Broker")
+		}
+	}
+
+	var repository string
+	var debug bool
+	var imageOverride map[string]string
+
+	if clusterInfo.Submariner != nil {
+		repository = clusterInfo.Submariner.Spec.Repository
+		imageOverride = clusterInfo.Submariner.Spec.ImageOverrides
+		debug = clusterInfo.Submariner.Spec.Debug
+	} else if clusterInfo.ServiceDiscovery != nil {
+		repository = clusterInfo.ServiceDiscovery.Spec.Repository
+		imageOverride = clusterInfo.ServiceDiscovery.Spec.ImageOverrides
+		debug = clusterInfo.ServiceDiscovery.Spec.Debug
+	}
+
+	// Upgrade Operator if deployed
+	if err := upgradeOperator(ctx, clusterInfo, repository, debug, imageOverride, status); err != nil {
+		return status.Error(err, "Error upgrading Operator")
+	}
+
+	// Upgrade Submariner
+	if clusterInfo.Submariner != nil {
+		status.Start("Found Submariner components. Upgrading it to %s", to)
+
+		clusterInfo.Submariner.Spec.Version = to
+
+		err := deploy.SubmarinerFromSpec(ctx, clusterInfo.ClientProducer.ForGeneral(), &clusterInfo.Submariner.Spec)
+		if err != nil {
+			return status.Error(err, "Error upgrading Submariner")
+		}
+
+		status.Success("Submariner successfully upgraded")
+	}
+
+	// Upgrade Service discovery
+	if clusterInfo.ServiceDiscovery != nil {
+		status.Start("Found Service Discovery components. Upgrading it to %s", to)
+
+		clusterInfo.ServiceDiscovery.Spec.Version = to
+
+		err := deploy.ServiceDiscoveryFromSpec(ctx, clusterInfo.ClientProducer.ForGeneral(), &clusterInfo.ServiceDiscovery.Spec)
+		if err != nil {
+			return status.Error(err, "Error upgrading Service Discovery")
+		}
+
+		status.Success("Service discovery successfully upgraded.")
+	}
+
+	return nil
+}
+
+func upgradeBroker(ctx context.Context, clusterInfo *cluster.Info, status reporter.Interface, namespace string,
+	brokerSpec v1alpha1.BrokerSpec,
+) error {
+	status.Start("Found Broker installed. Upgrading it to %s", to)
+	options := &deploy.BrokerOptions{
+		ImageVersion:    to,
+		BrokerNamespace: namespace,
+		BrokerSpec:      brokerSpec,
+	}
+
+	if err := deploy.Deploy(ctx, options, status, clusterInfo.ClientProducer); err != nil {
+		return err //nolint:wrapcheck // No need to wrap here
+	}
+
+	status.Success("Broker successfully upgraded.")
+
+	return nil
+}
+
+func upgradeOperator(ctx context.Context, clusterInfo *cluster.Info, repository string, debug bool, imageOverride map[string]string,
+	status reporter.Interface,
+) error {
+	status.Start("Checking if Operator is deployed")
```

Comment:
```suggestion
	status.Start("Checking if the Operator is installed")
```

Note that we use the word "installed" elsewhere so we should be consistent (or use the word "deployed" everywhere).